cluster/storage-cluster.go

```diff
@@ -0,0 +1,250 @@
+// This file is part of MinIO Kubernetes Cloud
+// Copyright (c) 2019 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+package cluster
+
+import (
+	"errors"
+	"fmt"
+)
+
+type StorageCluster struct {
+	Id   int32
+	Name *string
```

Comment:
```suggestion
	Name string
```

Reply:
Storage cluster name is optional, so the field may be `NULL` on the database.

Reply:
So, your saying that this struct may get marshaled into a SQL table entry? And this marshaling treats a nil pointer as `NULL` but `""`(empty string) not? If so, why don't we use the null-able types in the `sql` package? E.g. https://golang.org/pkg/database/sql/#NullString.

Also, if this (using `*string`) is necessary than we need comments explaining this since that is not obvious AFAICS.
E.g. `Name *string // Optional cluster name - nil pointer maps to NULL on DB`